portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,15 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+function projectToggle(id){
+    //the id for each button is proj[i], i being the index
+    //c-proj[i] is the id for the div with the description of each project
+
+    if($("#c-"+id)[0].style.display!="block"){
+        $("#c-"+id)[0].style.display="block"
+        location.href = "#c-"+id;
+    } else{
+          $("#c-"+id)[0].style.display="none"
+    };
+}
```

Comment:
Nice show/hide toggle!

Two possible improvements I could see:

- Use event delegation and bubbling on the projects list rather than inline `onclick` handlers on each project element. Great guide here: https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events

- Toggle a CSS class rather than directly setting style properties.  This can be a class toggled on the click event target element or you could query a different element to toggle the class on - kind of like you do here.

There's a really easy way to do both of these things in jQuery, but I'd challenge you to do it using plain JS.